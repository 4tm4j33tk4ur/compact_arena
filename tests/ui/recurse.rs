extern crate compact_arena;

use compact_arena::{mk_nano_arena, Idx8};

fn recursive(idx: Option<Idx8<'_>>) {
    mk_nano_arena!(arena); // `tag` does not live long enough
    if let Some(idx) = idx {
        assert_eq!("hello", arena[idx]);
    } else {
        recursive(Some(arena.add("hello")));
    }
}

fn main() {
    recursive(None);
}
