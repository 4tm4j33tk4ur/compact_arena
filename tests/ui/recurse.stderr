error[E0597]: `tag` does not live long enough
  --> $DIR/recurse.rs:6:5
   |
5  | fn recursive(idx: Option<Idx8<'_>>) {
   |              --- has type `std::option::Option<compact_arena::Idx<'1, u8>>`
6  |     mk_nano_arena!(arena); // `tag` does not live long enough
   |     ^^^^^^^^^^^^^^^^^^^^^^
   |     |
   |     borrowed value does not live long enough
   |     argument requires that `tag` is borrowed for `'1`
...
12 | }
   | - `tag` dropped here while still borrowed
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: aborting due to previous error

For more information about this error, try `rustc --explain E0597`.
